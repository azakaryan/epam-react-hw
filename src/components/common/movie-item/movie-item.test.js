import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import MovieItem from './movie-item';

expectEnzymify.extend(enzymify());

const getMovie = () => ({
  budget: 160000000,
  genres: ['Action', 'Thriller', 'Science Fiction', 'Mystery', 'Adventure'],
  id: 27205,
  overview:
    "Cobb, a skilled thief who commits corporate espionage by infiltrating the subconscious of his targets is offered a chance to regain his old life as payment for a task considered to be impossible: 'inception', the implantation of another person's idea into a target's subconscious.",
  poster_path:
    'https://image.tmdb.org/t/p/w500/qmDpIHrmpJINaRKAfWQfftjCdyi.jpg',
  release_date: '2010-07-14',
  revenue: 825532764,
  runtime: 148,
  tagline: 'Your mind is the scene of the crime.',
  title: 'Inception',
  vote_average: 8.2,
  vote_count: 16798,
});

describe('MovieItem Component', () => {
  it('should render correctly', () => {
    const component = shallow(<MovieItem getMovie={getMovie} />);

    expect(component).toMatchSnapshot();
  });

  it('renders a MovieList element', () => {
    const wrapper = shallow(<MovieItem getMovie={getMovie} />);
    // TODO how to use expect for both cases.
    expectEnzymify(wrapper).toBeA('Link');
  });
});
