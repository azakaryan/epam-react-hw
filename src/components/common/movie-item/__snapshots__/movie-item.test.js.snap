// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem Component should render correctly 1`] = `
<Link
  key="27205"
  to="/movie/27205"
>
  <div
    className="movie_item"
    id="movie"
  >
    <div
      className="image"
    >
      <img
        src="https://image.tmdb.org/t/p/w500/qmDpIHrmpJINaRKAfWQfftjCdyi.jpg"
      />
    </div>
    <div
      className="title"
    >
      <div>
        Inception
      </div>
      <div>
        2010
      </div>
    </div>
    <div
      className="genres"
    >
      Action & Thriller & Science Fiction & Mystery & Adventure
    </div>
  </div>
</Link>
`;
